client/src/templates/Challenges/odin/Show.tsx

```diff
@@ -0,0 +1,431 @@
+  data: { challengeNode: ChallengeNode };
+  description: string;
+  isChallengeCompleted: boolean;
+  openCompletionModal: () => void;
+  pageContext: {
+    challengeMeta: ChallengeMeta;
+  };
+  t: TFunction;
+  updateChallengeMeta: (arg0: ChallengeMeta) => void;
+  updateSolutionFormValues: () => void;
+}
+
+interface ShowOdinState {
+  subtitles: string;
+  downloadURL: string | null;
+  selectedOption: number | null;
+  answer: number;
+  showWrong: boolean;
+  assignmentsCompleted: number;
+  allAssignmentsCompleted: boolean;
+  videoIsLoaded: boolean;
+}
+
+// Component
+class ShowOdin extends Component<ShowOdinProps, ShowOdinState> {
+  static displayName: string;
+  private _container: HTMLElement | null | undefined;
+
+  constructor(props: ShowOdinProps) {
+    super(props);
+    this.state = {
+      subtitles: '',
+      downloadURL: null,
+      selectedOption: null,
+      answer: 1,
+      showWrong: false,
+      assignmentsCompleted: 0,
+      allAssignmentsCompleted: false,
+      videoIsLoaded: false
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount(): void {
+    const {
+      challengeMounted,
+      data: {
+        challengeNode: {
+          challenge: { title, challengeType, helpCategory }
+        }
+      },
+      pageContext: { challengeMeta },
+      updateChallengeMeta
+    } = this.props;
+    updateChallengeMeta({
+      ...challengeMeta,
+      title,
+      challengeType,
+      helpCategory
+    });
+    challengeMounted(challengeMeta.id);
+    this._container?.focus();
+  }
+
+  componentDidUpdate(prevProps: ShowOdinProps): void {
+    const {
+      data: {
+        challengeNode: {
+          challenge: { title: prevTitle }
+        }
+      }
+    } = prevProps;
+    const {
+      challengeMounted,
+      data: {
+        challengeNode: {
+          challenge: { title: currentTitle, challengeType, helpCategory }
+        }
+      },
+      pageContext: { challengeMeta },
+      updateChallengeMeta
+    } = this.props;
+    if (prevTitle !== currentTitle) {
+      updateChallengeMeta({
+        ...challengeMeta,
+        title: currentTitle,
+        challengeType,
+        helpCategory
+      });
+      challengeMounted(challengeMeta.id);
+    }
+  }
+
+  handleSubmit(
+    solution: number,
+    openCompletionModal: () => void,
+    assignments: string[]
+  ) {
+    const hasAssignments = assignments[0] != '';
+    const completed = this.state.allAssignmentsCompleted;
+
+    if (solution - 1 == this.state.selectedOption && hasAssignments) {
+      this.setState({
+        showWrong: false
+      });
+
+      if (completed) {
+        openCompletionModal();
+      }
+    } else if (solution - 1 === this.state.selectedOption && !hasAssignments) {
+      this.setState({
+        showWrong: false
+      });
+      openCompletionModal();
+    } else if (solution - 1 !== this.state.selectedOption) {
+      this.setState({
+        showWrong: true
+      });
+    }
```

Comment:
Also, thinking about it, it makes more sense to represent an absence of assignments with an empty array.  Then you can check for that rather than looking at the first element.